src/airdrop/src/lib.rs

```diff
@@ -139,7 +141,10 @@ fn add_codes(codes: Vec<String>) -> CustomResult<()> {
     // generate non activated codes
     mutate_state(|state| {
         for code in codes {
-            state.pre_generated_codes.push(Code(code));
+            // only add the code if it does not already exist
+            if !state.codes.contains_key(&Code(code.clone())) {
```

Comment:
Why not just using `insert`?

https://doc.rust-lang.org/stable/std/collections/struct.HashMap.html#method.insert